src/psc_bgk_util/params_parser.hxx

```diff
@@ -43,18 +43,37 @@ public:
   T get(const std::string paramName);
 
   template <typename T>
-  T getOrDefault(const std::string paramName, T deflt)
-  {
-    if (params.count(paramName) == 1)
-      return get<T>(paramName);
-    std::cout << "Warning: using default value for parameter '" << paramName
-              << "': " << deflt << std::endl;
-    return deflt;
-  }
+  T getOrDefault(const std::string paramName, T deflt);
+
+  // return true and display warning iff paramName is present
+  bool warnIfPresent(const std::string paramName, const std::string advice);
```

Comment:
Not really a useful comment at this time, but C++17 has `std::optional` (https://en.cppreference.com/w/cpp/utility/optional), which would be a standard way of handling something like this. But we're not requiring C++17 yet, and it's not worth it just for that.